server/state.go

```diff
@@ -0,0 +1,26 @@
+package server
+
+import (
+	"path/filepath"
+
+	"github.com/spacemeshos/poet/util"
+)
+
+const stateFilename = "state.bin"
+
+type state struct {
+	PrivKey []byte
+}
```

Comment:
I think the idea was that the state could be extended this way. It doesn't hurt to embed the key into a structure but it makes it more flexible.

Reply:
It only becomes more flexible if the file also contains some meta information about the data stored inside of it. Otherwise if we extend `state` in the future, there is no difference between having an object or just bytes.

With my comment I wanted to say we don't need XDR to encode a private key; `go-scale` would do the same thing but only use one byte instead of 4 to encode the length (and be faster in doing so), if we encode in hex it would be less efficient but human readable (the primary reason we encode in hex everywhere else).

Reply:
> It only becomes more flexible if the file also contains some meta information about the data stored inside of it. Otherwise if we extend state in the future, there is no difference between having an object or just bytes.

The meta information could be in the filename ;) i.e `state.bin`, `state_v1.bin` etc.

Anyway, don't get me wrong - I'm not defending it - you asked why is it a struct and that's my guess. This code has already been there and changing it in this PR is out of its scope. Personally, I'd migrate all the state files into JSON.